gst/nnstreamer/nnstreamer_plugin_api_util_impl.c

```diff
@@ -1087,6 +1083,27 @@ gst_tensor_dimension_is_valid (const tensor_dim dim)
   return TRUE;
 }
 
+/**
+ * @brief Compare the tensor dimension.
+ * @return TRUE if given tensors have same dimension.
+ */
+gboolean
+gst_tensor_dimension_is_equal (const tensor_dim dim1, const tensor_dim dim2)
+{
+  guint i;
+
+  /* Do not compare zero dimension. */
+  if (dim1[0] == 0U && dim2[0] == 0U)
+    return FALSE;
+
+  for (i = 0; i < NNS_TENSOR_RANK_LIMIT; i++) {
+    if (dim1[i] != dim2[i])
+      return FALSE;
+  }
```

Comment:
with 0-init, we may have 4:3:1:00000 and 4:3:000000. Are they different? or same?

Reply:
1. After implementing 0-init dims, 4:3:1:00 and 4:3:00 will be same.
2. Current dimension is filled with 1, and this util function should be updated later.
e.g., compare remained dim like below, (I didnt add below lines in this PR.)
```
    if (dim1[i] != dim2[i]) {
      tensor_dim remained;

      if (dim1[i] == 0)
        remained = dim2;
      else if (dim2[i] == 0)
        remained = dim1;
      else
        return FALSE;

      /* Supposed dimension is same if remained dimension is 1. */
      for (; i < NNS_TENSOR_RANK_LIMIT; i++) {
        if (remained[i] > 1)
          return FALSE;
      }
    }
```